controls/src/main/java/com/aol/mobile/sdk/controls/view/AdsSeekBar.java

```diff
@@ -0,0 +1,77 @@
+package com.aol.mobile.sdk.controls.view;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.drawable.Drawable;
+import android.graphics.drawable.LayerDrawable;
+import android.util.AttributeSet;
+import android.widget.SeekBar;
+
+import com.aol.mobile.sdk.controls.R;
+
+import java.util.Set;
+
+public class AdsSeekBar extends SeekBar {
+
+    private Paint midrollPaint = new Paint();
+    private Set<Float> midrolls;
+    private int width;
+    private int height;
+    private float radius;
+
+    public AdsSeekBar(final Context context) {
+        this(context, null);
+    }
+
+    public AdsSeekBar(final Context context, final AttributeSet attrs) {
+        this(context, attrs, 0);
+    }
+
+    public AdsSeekBar(final Context context, final AttributeSet attrs, final int defStyle) {
+        super(context, attrs, defStyle);
+        midrollPaint.setFlags(Paint.ANTI_ALIAS_FLAG);
+        midrollPaint.setColor(getResources().getColor(R.color.default_ad_accent_color));
+        updateMarkParams();
+    }
+
+    public void setMidrolls(Set<Float> midrolls) {
+        boolean needUpdate = !midrolls.equals(this.midrolls);
```

Comment:
`needUpdate` will always be `true` because `vm.midrolls` is single final `HashMap`. This needs rework.